src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,83 @@
+package duke.task;
+
+import duke.storage.CSV;
+import duke.storage.CSVParsable;
+
+public abstract class Task implements CSVParsable {
+    private String name;
+    protected boolean done = false;
+    protected TaskType type;
+
+    protected Task(String name, TaskType type) {
+        this.name = name;
+        this.type = type;
+    }
+
+    /**
+     * getter for done
+     *
+     * @return done
+     */
+    public boolean isDone() {
+        return this.done;
+    }
+
+    /**
+     * set done to true
+     *
+     * @return true
+     */
+    public boolean setToDone() {
+        return this.done = true;
+    }
+
+    /**
+     * getter for name
+     *
+     * @return name
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * getter for task type
+     *
+     * @return type
+     */
+    public TaskType getType() {
+        return this.type;
+    }
+
+    protected String gou() {
+        return isDone() ? "✓" : "✗";
+    }
+
+    protected String sqB(String s) {
+        return "[" + s + "]";
+    }
+
+    public CSV toCSV() {
+        return new CSV(new CSV("T"), new CSV(Boolean.toString(isDone())), new CSV(getName()));
+    }
+
+    /**
+     * Parse tasks from csv object
+     *
+     * @param csv = csv that described the task (parsed from local file)
+     * @return previously saved task
+     */
+    public static Task parseFromCSV(CSV csv) {

```

Comment:
The parent should not have knowledge about the child. Instead, it should define or handle only generic behaviors. There is no need for having this parsing within the Task class. You can incorporate this behavior in any parsing or storage-related classes.